src/tools/fuzzing/fuzzing.cpp

```diff
@@ -746,6 +747,37 @@ void TranslateToFuzzReader::finalizeTable() {
   }
 }
 
+void TranslateToFuzzReader::shuffleExports() {
+  // Randomly ordering the exports is useful for a few reasons. First, initial
+  // content may have a natural order in which to execute things (an "init"
+  // export first, for example), and changing that order may lead to very
+  // different execution. Second, even in the fuzzer's own random content there
+  // is a "direction", since we generate as we go (e.g. no function calls a
+  // later function that does not exist yet / will be created later), and also
+  // we emit invokes for a function right after it (so we end up calling the
+  // same code several times in succession, but interleaving it with others may
+  // find more things). But we also keep a good chance for the natural order
+  // here, as it may help some initial content.
+  if (wasm.exports.empty() || oneIn(2)) {
+    return;
+  }
+
+  // Sort the exports in the simple Fisher-Yates manner.
+  // https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle#The_modern_algorithm
+  for (Index i = 0; i < wasm.exports.size() - 1; i++) {
```

Comment:
The last item is uniquely determined by all the previously picked items.
```suggestion
  for (Index i = 0; i < wasm.exports.size() - 2; i++) {
```

Reply:
True, but then we need to guard against the exports being of size 1? I guess I can make the earlier check do that, but I'm not sure if it's clearer, what do you think?

Reply:
Oh yeah, good point. Another fix would be to use a signed type for the index variable. I don't feel strongly about whether either of those options is better than the current code. I guess I would probably modify the check like you mentioned.

Reply:
I think that's right. Sorry for the confusion!